apps/permissions/src/app/guards.ts

```diff
@@ -0,0 +1,27 @@
+import { inject } from '@angular/core';
+import { Router } from '@angular/router';
+import { map } from 'rxjs';
+import { Role } from './user.model';
+import { UserStore } from './user.store';
+
+export const isAdmin = () => {
+  const userStore = inject(UserStore);
+  const router = inject(Router);
+  return userStore.isAdmin$.pipe(
+    map((isAdmin) => {
+      return isAdmin ? true : router.createUrlTree(['/access-denied']);
+    })
+  );
+};
+
+export const hasPermissions = (
+  roles: Role[],
+  userStore = inject(UserStore)
+) => {
+  const router = inject(Router);
```

Comment:
I think it's better to place every injectable service inside the parameter of your function (same as `userStore`) with a default value. 

- easier to test
- can pass services when used outside the injection context